Source/JavaScriptCore/runtime/StringConstructor.cpp

```diff
@@ -89,15 +89,15 @@ JSC_DEFINE_HOST_FUNCTION(stringFromCharCode, (JSGlobalObject* globalObject, Call
         return JSValue::encode(jsSingleCharacterString(vm, code));
     }
 
-    LChar* buf8Bit;
+    std::span<LChar> buf8Bit;
     auto impl8Bit = StringImpl::createUninitialized(length, buf8Bit);
     for (unsigned i = 0; i < length; ++i) {
         UChar character = static_cast<UChar>(callFrame->uncheckedArgument(i).toUInt32(globalObject));
         RETURN_IF_EXCEPTION(scope, encodedJSValue());
         if (UNLIKELY(!isLatin1(character))) {
-            UChar* buf16Bit;
+            std::span<UChar> buf16Bit;
             auto impl16Bit = StringImpl::createUninitialized(length, buf16Bit);
-            StringImpl::copyCharacters(buf16Bit, { buf8Bit, i });
+            StringImpl::copyCharacters(buf16Bit.data(), buf8Bit.first(i));
```

Comment:
Yes, I noticed `copyCharacters` but opted to not to that larger refactoring in this patch. I agree though that copyCharacters() would ideally take 2 spans instead of 1.